openedx/core/djangoapps/credentials/tasks/v1/tasks.py

```diff
@@ -54,3 +62,178 @@ def send_grade_to_credentials(self, username, course_run_key, verified, letter_g
             f"Failed to send grade to credentials. Reason: {error_msg}"
         )
         raise self.retry(exc=exception, countdown=countdown, max_retries=MAX_RETRIES)
+
+
+@shared_task(bind=True, ignore_result=True)
+@set_code_owner_attribute
+def handle_notify_credentials(self, options, course_keys):
+    """
+    Celery task to handle the notify_credentials management command. Finds the
+    relevant cert and grade records, then starts other celery tasks to send the
+    data.
+    """
+
+    try:
+        site_config = SiteConfiguration.objects.get(site__domain=options['site']) if options['site'] else None
+    except SiteConfiguration.DoesNotExist:
+        logger.error('No site configuration found for site %s', options['site'])
+
+    certs = get_recently_modified_certificates(
+        course_keys, options['start_date'], options['end_date'], options['user_ids']
+    )
+
+    users = None
+    if options['user_ids']:
+        users = User.objects.filter(id__in=options['user_ids'])
+
+    grades = get_recently_modified_grades(
+        course_keys, options['start_date'], options['end_date'], users
+    )
+
+    logger.info('notify_credentials Sending notifications for {certs} certificates and {grades} grades'.format(
+        certs=certs.count(),
+        grades=grades.count()
+    ))
+
+    if options['dry_run']:
+        print_dry_run(certs, grades)
+    else:
+        send_notifications(
+            certs,
+            grades,
+            site_config=site_config,
+            delay=options['delay'],
+            page_size=options['page_size'],
+            verbose=options['verbose'],
+            notify_programs=options['notify_programs']
+        )
+
+    logger.info('notify_credentials finished')
+
+
+def send_notifications(
+    certs, grades, site_config=None, delay=0, page_size=100, verbose=False, notify_programs=False
```

Comment:
I’ve changed the `page_size` default here from `0` to `100`. 100 matches the default given at the flag level ([see here](https://github.com/edx/edx-platform/blob/3660b2a34619a0f23ddc28578eef08f2e69dafde/openedx/core/djangoapps/credentials/management/commands/notify_credentials.py#L155)). Besides, `0` (if it was every actually inserted here—which I don’t think it ever would be) would cause problems later on because we divide the total results by `page_size`, and dividing by zero is bad news.